src/factory/crawler-factory.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Question on filenaming convention.  I can see the PascalCasing for files that export a class.  IMHO camelCase should be used for all others.  '-' just gets cumbersome.  Is there a particular convention to which you are appealing?  This file should be crawlerFactory.js.  Similarly for its siblings.